.gitlab-ci.yml

```diff
@@ -0,0 +1,102 @@
+variables:
+  DOCKER_REPOSITORY: "mozmeao/basket"
+  DOCKER_IMAGE_TAG: "${DOCKER_REPOSITORY}:${CI_COMMIT_SHORT_SHA}"
+  # Cannot use $DOCKER_IMAGE_TAG because it contains other variables itself.
+  DOCKER_RUN_CMD: "docker run --env-file .docker-env ${DOCKER_REPOSITORY}:${CI_COMMIT_SHORT_SHA}"
+
+stages:
+  - build
+  - test
+  - deploy
+
+build-images:
+  stage: build
+  tags:
+    - meao
+  script:
+    - docker build -t ${DOCKER_IMAGE_TAG} --build-arg GIT_SHA=${CI_COMMIT_SHA} --pull .
+    - docker push ${DOCKER_IMAGE_TAG}
+
+.test-stage:
+  stage: test
+  variables:
+    DOCKER_DEBUG: "False"
+    DOCKER_ALLOWED_HOSTS: "*"
+    DOCKER_SECRET_KEY: "foo"
+    DOCKER_DATABASE_URL: "sqlite:///"
+    DOCKER_SITE_URL: "http://localhost:8000"
+    DOCKER_CACHE_URL: "dummy://"
+    DOCKER_ENABLE_ADMIN: "True"
+    DOCKER_SECURE_SSL_REDIRECT: "False"
+  before_script:
+    - env | grep "^DOCKER" | sed 's/DOCKER_//' > .docker-env
+
+check-migrations:
+  extends: .test-stage
+  tags:
+    - meao
+  script:
+    - ${DOCKER_RUN_CMD} bash -c './manage.py makemigrations  | grep "No changes detected"'
+
+run-flake8:
+  extends: .test-stage
+  tags:
+    - meao
+  script:
+    - ${DOCKER_RUN_CMD} flake8 snippets
+
+run-unit-tests:
+  extends: .test-stage
+  tags:
+    - meao
+  coverage: '/TOTAL.+ ([0-9]{1,3}%)/'
+  script:
+    - ${DOCKER_RUN_CMD} bash -c 'coverage run ./manage.py test --parallel; coverage report'
```

Comment:
this differs from the test command in `run-tests.sh`:

`py.test --junitxml=test-results/test-results.xml basket`

Reply:
We decided not to use junit xml for gitlab; this was used by a Jenkins plugin.